dotnet/Liberator.Driver/Performance/RatWatch.cs

```diff
@@ -112,48 +123,68 @@ public double ArithmeticMean()
         /// <summary>
         /// Calculates the geometric mean for the timings list
         /// </summary>
+        /// <param name="enumTiming">Type of timing point</param>
         /// <returns>The geometic mean for the timings list</returns>
-        public double GeometricMean()
+        public double GeometricMean([DefaultParameterValue(EnumTiming.NotSpecified)]EnumTiming enumTiming)
         {
             try
             {
-                return Statistics.GeometricMean(_valueList);
+                IEnumerable<double> valueList =
+                    Timings.Where(v => v.Item1 == enumTiming)
+                    .DefaultIfEmpty()
+                    .Select(v => v.Item2.Duration.TotalMilliseconds);
+
+                return Statistics.GeometricMean(valueList);
```

Comment:
what is the difference between the Geometric and Statistical mean methods?

Reply:
don't worry I've seen it now

Reply:
🥇